### PR TITLE
mnist_hogwild manual breaking of gradient sharing removed

### PR DESCRIPTION
**Pull request by pfre00**
Since gradients are allocated lazily, they are instantiated in every process separately on the `loss.backward()` call. The removed lines in train.py are no longer needed.

### CODE REVIEW

**Comment by apaszke** [comment]
Thank you!
